scripts/app.js

```diff
@@ -1,39 +1,55 @@
-var youtube_search = 'https://www.googleapis.com/youtube/v3/search';
 
-function getDataFromApi(searchTerm, callback) {
-  var query = {
+// This is the YouTube API call Object. It also has a prototypical inheritance of the 'callAPI' method
+var Data = function(searchTerm, callback){
+  this.url = 'https://www.googleapis.com/youtube/v3/search';
+  this.query = {
     part: 'snippet',
-    q: searchTerm,    
+    q: searchTerm,
     key: 'AIzaSyBN_2IeT7Ez2Wp2gsecionVgTUc4uL-hh4',
   }
-  $.getJSON(youtube_search, query, callback);
+  console.log(typeof(Data.callAPI));
 }
 
 
+// The API call to YouTube
+Data.prototype.callAPI = function(){
+  $.getJSON(url, query, callback);
```

Comment:
You could either define a `callback` function outside this function and then pass it as an argument, or you could pass an anonymous function in place of `callback`. Like:

```
$.getJSON(url, query, function(data) {
// stuff to do when successful
});
```